bin/glm

```diff
@@ -0,0 +1,253 @@
+        }
+        if (labelsToDelete.length !== 0) {
+          await deleteLabels(labelsToDelete);
+        }
+        if (labelsToUpdate.length !== 0) {
+          await updateLabels(labelsToUpdate);
+        }
+        const changes =
+          newLabels.length + labelsToDelete.length + labelsToUpdate.length;
+        const summary =
+          changes !== 0
+            ? {
+                created: newLabels.length,
+                deleted: labelsToDelete.length,
+                updated: labelsToUpdate.length,
+              }
+            : 'Labels already up to date!';
+        resolve(summary);
+      });
+    } catch (err) {
+      reject(err);
+    }
+  });
+}
+
+function checkForNewOrOldLabels(newLabels, oldLabels) {
+  return new Promise((resolve, reject) => {
+    try {
+      const updatedLabels = [];
+      const OldLabelIds = [];
+      oldLabels.forEach(label => {
+        OldLabelIds.push(label.id ? label.id : undefined);
+      });
+      newLabels.forEach(label => {
+        if (!OldLabelIds.includes(label.id ? label.id : undefined)) {
+          updatedLabels.push(label);
+        }
+      });
+      resolve(updatedLabels);
+    } catch (err) {
+      reject(err);
+    }
+  });
+}
+
+function checkForModifications(localConfig, publishedConfig) {
+  return new Promise(async (resolve, reject) => {
+    const formattedPublishedConfig = await labelFormatter(publishedConfig);
+    const labelChanges = [];
+    try {
+      localConfig.forEach(localLabel => {
+        formattedPublishedConfig.forEach(publicLabel => {
+          if (localLabel.id === publicLabel.id) {
+            if (
+              Object.values(localLabel).toString() !==
+              Object.values(publicLabel).toString()
+            ) {
+              labelChanges.push(localLabel);
+            }
+          }
+        });
+      });
+      resolve(labelChanges);
+    } catch (err) {
+      reject(err);
+    }
+  });
+}
+
+function labelFormatter(labelData) {
+  return new Promise((resolve, reject) => {
+    try {
+      const formattedLabels = [];
+      labelData.forEach(label => {
+        const data = {
+          id: label.id,
+          name: label.name,
+          color: label.color,
+          description: label.description,
+        };
+        formattedLabels.push(data);
+      });
+      resolve(formattedLabels);
+    } catch (err) {
+      reject(err);
+    }
+  });
+}
+
+function fetchLabels() {
+  const url = `https://api.github.com/repos/${owner}/${repo}/labels`;
+  return axios.get(url).then(response => {
+    const labels = [];
+    response.data.forEach(label => {
+      labels.push(label);
+    });
+    return labels;
+  });
+}
+
+function createLabels(labels) {
+  const url = `https://api.github.com/repos/${owner}/${repo}/labels`;
+  labels.forEach(label => {
+    const { name } = label;
+    const { description } = label.description || '';
+    const { color } = label.color || 'ededed';
+    return axios
```

Comment:
There are several things wrong here.
- you can't return inside a `forEach`
- you can't control async stuff inside a `forEach`, at least not like this
- the function `createLabels` does not have any return value like this, so if you `await` on it it does not work

What you want is to **return** a `Promise.all`

Reply:
👍  567f587